discovery_engine_core/core/src/stack/ops/breaking.rs

```diff
@@ -44,34 +42,31 @@ impl Ops for BreakingNews {
     }
 
     fn configure(&mut self, config: &EndpointConfig) {
+        self.client = Client::new(config.api_key.clone(), config.api_base_url.clone());
         self.markets.replace(Arc::clone(&config.markets));
     }
 
-    #[allow(clippy::cast_precision_loss)]
-    #[allow(clippy::cast_possible_truncation)]
     async fn new_items(&self, _key_phrases: &[KeyPhrase]) -> Result<Vec<Article>, GenericError> {
-        let n = 10;
-        let articles = (0..n).fold(Vec::with_capacity(n), |mut articles, i| {
-            articles.push(
-            Article {
-                id: i.to_string(),
-                title: format!("B Document Title {}", i),
-                score: if i % 2 == 0 {Some(i as f32) } else {None},
-                rank: i,
-                source_domain: "xayn.com".to_string(),
-                excerpt: format!("Content of the news {}", i),
-                link: "https://xayn.com/".into(),
-                media: "https://uploads-ssl.webflow.com/5ea197660b956f76d26f0026/614349038d7d72d1576ae3f4_plant.svg".into(),
-                topic: Topic::Unrecognized,
-                country: "DE".to_string(),
-                language: "de".to_string(),
-                published_date: NaiveDate::from_ymd(2022, 2, (i + 1) as u32).and_hms(9, 10, 11),
-            });
-
-            articles
-        });
+        if let Some(markets) = self.markets.as_ref() {
+            let mut articles = Vec::new();
+            let mut errors = Vec::new();
+            let page_size = Some(20); // TODO pass through config later
 
-        Ok(articles)
+            for market in markets.read().await.clone() {
+                let query = HeadlinesQuery { market, page_size };
+                match self.client.headlines(&query).await {
+                    Ok(batch) => articles.extend(batch),
+                    Err(err) => errors.push(err),
+                }
+            }
+            if articles.is_empty() && !errors.is_empty() {
```

Comment:
I'm not sure about `articles.is_empty()` that is not really an error, it could happen that the api returns 0 documents.
Will be weird for the headlines and kinda possible for the personalized news if the keyword don't have a match.

I think we can merge and return on this in another pr if we want to remove it.

Reply:
>I'm not sure about `articles.is_empty()` that is not really an error, it could happen that the api returns 0 documents.

yeah. if the api returns no articles and there were no errors throughout, that's definitely an ok. my thinking was if there are no articles but there _are_ errors, returning an error would be more informative. but I see your point if returning 0 articles is a very possible / frequent occurrence e.g. for personalized news.

ok will merge as is but if you prefer it changed I will follow it up.